libs/langgraph/langgraph/prebuilt/tool_node.py

```diff
@@ -391,23 +425,78 @@ def __init__(self, field: Optional[str] = None) -> None:
         self.field = field
 
 
+class InjectedStore(InjectedToolArg):
+    """Annotation for a Tool arg that is meant to be populated with LangGraph store.
+
+    Any Tool argument annotated with InjectedStore will be hidden from a tool-calling
+    model, so that the model doesn't attempt to generate the argument. If using
+    ToolNode, the appropriate store field will be automatically injected into
+    the model-generated tool args. Note: if a graph is compiled with a store object,
+    the store will be automatically propagated to the tools with InjectedStore args
+    when using ToolNode.
+
+    Example:
+        ```python
+        from typing import Any
+        from typing_extensions import Annotated
+
+        from langchain_core.messages import AIMessage
+        from langchain_core.tools import tool
+
+        from langgraph.store.memory import InMemoryStore
+        from langgraph.prebuilt import InjectedStore, ToolNode
+
+        store = InMemoryStore()
+        store.put(("values",), "foo", {"bar": 2})
+
+        @tool
+        def store_tool(x: int, my_store: Annotated[Any, InjectedStore()]) -> str:
```

Comment:
Does it have to be `Any`?
Thoughts on it being

```
Annotated[BaseStore,  InjectedToolArg]
```

(so we don't require duplicate annotation type things?)

Reply:
let me improve

Reply:
there is an issue in `core`, will update once fixed up

Reply:
🉑